src/packages/next/lib/api/schema/compute/create-server.ts

```diff
@@ -0,0 +1,56 @@
+import { z } from "../../framework";
+
+import { FailedAPIOperationSchema } from "../common";
+
+import { ProjectIdSchema } from "../projects/common";
+
+import {
+  ComputeServerCloudSchema,
+  ComputeServerColorSchema,
+  ComputeServerIdBodySchema,
+  ComputeServerTitleSchema,
+  ServerConfigurationSchema
+} from "./common";
+
+// OpenAPI spec
+//
+export const CreateServerInputSchema = z
+  .object({
+    configuration: ServerConfigurationSchema,
+    title: ComputeServerTitleSchema,
+    color: ComputeServerColorSchema,
+    cloud: ComputeServerCloudSchema,
+    project_id: ProjectIdSchema
+      .describe(
+        "The project id that this compute server provides compute for."
+      ),
+    idle_timeout: z
+      .number()
+      .describe(
+        `The idle timeout in seconds of this compute server. If set to 0, the server will
+        never turn off automatically. The compute server idle timeouts if none of the tabs 
+        it is providing are actively touched through the web UI. _Not yet implemented._`
+      )
+      .optional(),
+    autorestart: z.boolean()
+      .describe(
+        `If true and the compute server stops for any reason, then it 
+        will be automatically started again. This is primarily useful for 
+        stopping instances.`
```

Comment:
This should say "This is primarily useful for spot instances.`"  You were copying from packages/util/db-schema/compute-servers.ts which had a different typo "This is primarily useful for stop instances." but you fixed it in the wrong way.

Reply:
Incidentally, you could through Javascript directly import and use the text from the files in util/db-schema.  It's all in one big javascript object "SCHEMA".

Reply:
Whoops :sweat_smile: Thanks for the feedback!

> Incidentally, you could through Javascript directly import and use the text from the files in util/db-schema. It's all in one big javascript object "SCHEMA".

At one point, I'd thought that it might be better if the schemas were more or less isolated from the rest of the codebase (since they effectively define an interface) to keep concerns separate, but this could be a pretty good way to maintain the whole "single source of truth" paradigm. I'm honestly not sure which is the better way to go here, so if you have a preference I'd be happy to run with that.

Reply:
Since in both cases we're 100% documenting a field (with no impact except documentation), and this very example illustrates how such documentation of the exact same thing can diverge and become inconsistent, I would say the code should clearly be the same.    I do this sort of refactoring constantly - that's what a lot of the current util package contains, is code  or data that should be used in multiple places.

I don't mean to replicate too much.  But documentation of a field is certainly fair game.  The typing schema is different/dangerous/tricky though (e.g., one is oriented toward postgres) so I wouldn't touch that.